README.md

```diff
@@ -104,7 +70,37 @@ case fetch_user(1) do
 end
 ```
 
-[Forum discussion on `with` naming](https://elixirforum.com/t/alternative-to-with-specific-to-result-tuples/3264)
+####  Error Matching
```

Comment:
Do you think this needs to be a separate section? Again it's probably separate because I added it later. I really like your three bullet points in basic usage maybe we could think of a 4th that summed up the error behaviour and add an error handler to the cart example?